crates/core/src/host/instance_env.rs

```diff
@@ -543,29 +539,25 @@ mod test {
 
     /// An `InstanceEnv` requires `ModuleSubscriptions`
     fn subscription_actor(relational_db: Arc<RelationalDB>) -> ModuleSubscriptions {
-        ModuleSubscriptions::new(
-            relational_db,
-            Arc::new(RwLock::new(SubscriptionManager::default())),
-            Identity::ZERO,
-        )
+        ModuleSubscriptions::new(relational_db, <_>::default(), Identity::ZERO)
     }
 
     /// An `InstanceEnv` requires a `ReplicaContext`.
     /// For our purposes this is just a wrapper for `RelationalDB`.
     fn replica_ctx(relational_db: Arc<RelationalDB>) -> Result<ReplicaContext> {
-        Ok(ReplicaContext {
-            database: Database {
+        Ok(ReplicaContext::new(
+            Database {
                 id: 0,
                 database_identity: Identity::ZERO,
                 owner_identity: Identity::ZERO,
                 host_type: HostType::Wasm,
                 initial_program: Hash::ZERO,
             },
-            replica_id: 0,
-            logger: Arc::new(temp_logger()?),
-            subscriptions: subscription_actor(relational_db.clone()),
+            0,
+            Arc::new(temp_logger()?),
+            subscription_actor(relational_db.clone()),
             relational_db,
-        })
+        ))
```

Comment:
NIT: Seems like a change for the worse to me. Now the `replica_id` is a magic number `0`.